core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java

```diff
@@ -726,6 +729,34 @@ public static String getOncoKBApiUrl()
         return "";
     }
 
+    public static String getCivicUrl()
+    {
+        String civicUrl = properties.getProperty(CIVIC_URL);
+        boolean showCivic = showCivic();
+
+        // This only applies if there is no civic.url property in the portal.properties file.
+        if (civicUrl == null || civicUrl.isEmpty()) {
+            civicUrl = "https://civic.genome.wustl.edu/api/";
+        }
+
+        //Test connection of CIVIC website.
+        if(!civicUrl.isEmpty() && showCivic) {
+
+            try {
+                URL url = new URL(civicUrl + "genes");
```

Comment:
btw: I would prefer this method to just return the configured URL and not test it. Perhaps add another method that tests it, so other code can decide to call this "test url" method if needed. I would prefer to see proper exception handling and logging here. E.g. if there is a problem, throw an exception and let other parts of the code catch. When throwing an exception, also call `log.error` with the Exception error message so that it appears in the cbioportal log files and an administrator can troubleshoot.